### ISSUE TITLE
ping.php returns read notifications

### ISSUE THREAD

**Issue by Extarys** [friendica/friendica#9797]
### Is the feature request related to a problem? Please describe.

`ping.php` is called every ~40 seconds. I noticed that it returns an array for the notifications (both read and unread). On my nerdica account, I have 5 notifications, all have been read. Still, the ping returns an array with all 5:
(Truncated with a single notification, just to show)
```
{
	"result": {
		"intro": 0,
		"mail": 0,
		"net": 3,
		"home": 0,
		"register": 0,
		"all-events": 0,
		"all-events-today": 0,
		"events": 0,
		"events-today": 0,
		"birthdays": 0,
		"birthdays-today": 0,
		"groups": [],
		"forums": [],
		"notification": 0,
		"notifications": [
			{
				"id": 120603,
				"hash": "",
				"name": "Wolle",
				"url": "https://hasestall.com/profile/wolle1",
				"photo": "https://nerdica.net/photo/5310067931600198f1d8df9896626836-4.jpg?ts=1610717429",
				"date": "9 minutes ago",
				"msg": "Wolle commented on their post \"!Friendica Support Hey everybody! On my server, pictures posted by …\"",
				"uid": 3508,
				"link": "https://nerdica.net/display/9ac8db7c-2460-01e0-50df-c4d847420208",
				"parent": 46730987,
				"seen": 1,
				"type": 8,
				"verb": "http://activitystrea.ms/schema/1.0/post",
				"otype": "item",
				"iid": 46757660,
				"name_cache": "Wolle",
				"msg_cache": "{0} commented on their post \"!Friendica Support Hey everybody! On my server, pictures posted by …\"",
				"uri-id": 19324835,
				"parent-uri-id": 19316151,
				"visible": 1,
				"deleted": 0,
				"message": "{0} commented on their post \"!Friendica Support Hey everybody! On my server, pictures posted by …\"",
				"href": "https://nerdica.net/notification/120603",
				"timestamp": "2021-01-15 18:35:08"
			}
		],
		"sysmsgs": {
			"notice": [],
			"info": []
		}
	}
}
```
Just for fun, here's how I see it (yeah I'm weird):

**Every 40 seconds, for each client (not user, client)**

- 2.1 kb of data is transferred, for 5 notifications.
- You have 1 to 100 users on the server? Let's say 100 :) 210 kb every 40 seconds
- Every object in the notification is grabbed from the database (Takes CPU, Memory, Energy, Time)
- message and msg_cache are shortened (...) (Takes CPU, Memory, Energy, Time)
- This is sent to the client through Ajax (Bandwidth concern for the server AND the client, which may use a slow phone that require battery usage and bandwidth of a limited 2G/3G/XG plan)
- The browser itself need to rewrite the object and probably re-render the div (don't know didn't check but maybe)

It's not much on it's own, on my little tab, but it's still something that is worth addressing eventually.

### Describe the solution you'd like

- Fetch all notifications when the client open
  GET | https://nerdica.net/ping?format=json&uid=1234&inc_seen=1&_=1610737376384

- Recurring ping would only return new notifications.
   GET | https://nerdica.net/ping?format=json&uid=1234&_=1610737376384

`inc_seen=1` could be just `include=seen` (comma separated field)

I'm not sure if the current web client fetch the notification once when the page load, but having the flexibility to fetch all notifications is still worth having to display older ones. I took a quick look at `ping.php`, I _might_ try to fix this myself if you think this is worth doing, but  it's not a priority.

**_My conception of a ping is that it should only return new stuff._**

### Additional context

Long term solution could be to implement a graphql endpoint, and the client can decide what to fetch.

**Comment by MrPetovan** [comment]
Thank you for this suggestion. To answer your points, the notifications aren't loaded with the page itself, but we do call the ping routine once the DOM is ready to fetch the full list of notifications. From then on, the full list is used to update the display of read notifications and receive new unread ones, so there is room for improvement from this naive but robust implementation.
